code/game/machinery/camera/camera_assembly.dm

```diff
@@ -129,8 +129,13 @@
 			user << "The assembly already has \a [W] inside!"
 			return
 		user << "You attach the [W] into the assembly inner circuits."
-		upgrades += W
-		user.drop_item(W, src)
+		if (istype(W, /obj/item/stack/sheet/mineral/plasma))
+			var/obj/item/stack/sheet/mineral/plasma/s = W
+			s.use(1)
+			upgrades += new /obj/item/stack/sheet/mineral/plasma
+		else
+			upgrades += W
+			user.drop_item(W, src)

```

Comment:
Can you pretty please put this part at the beginning and have nothing happen in case drop_item() returns 0? Also might be good checking for /obj/item/stack instead of /obj/item/stack/sheet/mineral/plasma for an easier time in the future